### DISCUSSION TITLE
Welcome to Bard-API Discussions!

### DISCUSSION THREAD

**Discussion by dsdanielpark** [dsdanielpark/Bard-API#44] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Yazeed-Alharbi** [comment]
Is there anyway to manually add text to the conversation context? Just like the official GPT 3.5 turbo api, it allows you to pass a list of text ( the context ) with each request.

**Reply by dsdanielpark** [reply]
> Is there anyway to manually add text to the conversation context? Just like the official GPT 3.5 turbo api, it allows you to pass a list of text ( the context ) with each request.

If you are looking for prompt commands for specific roles, such as the role of "guide" in GPT, it is currently not officially supported by the Google BARD, but it is expected to be updated in the future. However, in reality, you can achieve a similar functionality by issuing commands in the next bard prompt or gpt prompt, where some prompts have been condensed on the backend.

While it is programmatically possible to implement this (by simply creating a dictionary and providing it), I'm not sure how many people would use it.

If there is significant demand for this, I could potentially implement it myself.

Before that, please refer to the GitHub repository for the amazing bard prompts.

https://github.com/dsdanielpark/amazing-bard-prompts

**Comment by Laura-Cuesta** [comment]
Hello!!! I'am Laura.
I'm a junior data scientist/engineer :-) and I was experimenting with your code.
Initially, I had problems with my token, but I changed the way (now I do bard= Bard(token=key)) and it is ok.
Then I haved problems with this asnwer:
"SNlM0e value not found in response. Check __Secure-1PSID value."

So i accessed to value of "resp.text" and I noticed that "SNlM0e" does'nt appear in this variable.

But now, when Itried again and I only get this error with the following code:

input_text = 'what is google bard'
bard= Bard(token=key)
bard.get_answer(input_text)['content']

'Response Error: b\')]}\\\'\\n\\n38\\n[["wrb.fr",null,null,null,null,[9]]]\\n56\\n[["di",55],["af.httprm",55,"-8600792505187596027",13]]\\n25\\n[["e",4,null,null,131]]\\n\'.'


Could you help me please?!

Thank you so much

**Reply by dsdanielpark** [reply]
I apologize for the delay in reviewing the mentioned content. Have you found a solution to the related issue in #1? We are currently rechecking the matter due to changes in the interface policy of the Google API. If you report the issue, we will kindly assist you. Wishing for a swift resolution and have a great day!

Regards,
Daniel

**Reply by lilbaby4** [reply]
I tried to install hhfsllm 
I am running python 3.8 on Windows 7.
i am unable to get torch installed.
python -m pip install torch>=1.12.0
No match distribution found for torch.

**Reply by dsdanielpark** [reply]
In fact, hf-transllm requires you to have Hugging Face's transformers installed on a system with suitable CUDA-enabled GPU resources or CPU resources along with Torch.
https://pytorch.org/get-started/locally/

 If you have a virtual environment with Torch properly installed, I recommend trying to run it within that environment. Setting up Torch and CUDA can vary greatly depending on the environment, so I cannot provide in-depth assistance, but you should be able to find plenty of resources through the PyTorch community or by searching online.

Once properly installed, hf-transllm offers the advantage of running seamlessly on-premises, although it may require significant storage capacity and resources.

**Comment by dsdanielpark** [comment]
> I tried to install hhfsllm I am running python 3.8 on Windows 7. i am unable to get torch installed. python -m pip install torch>=1.12.0 No match distribution found for torch.

https://pytorch.org/get-started/locally/
